server/lib/version.js

```diff
@@ -43,19 +37,16 @@ function getCommitHash () {
     /* ignore, shell out to `git` for hash */
   }
 
-  return new Promise(function (resolve) {
-    const gitDir = path.resolve(__dirname, '..', '..', '.git');
-    cp.exec('git rev-parse HEAD', { cwd: gitDir }, function (err, stdout) {
-      if (err) {
-        // ignore the error
-        resolve(UNKNOWN);
-        return;
-      }
+  let stdout = UNKNOWN;
+  const gitDir = path.resolve(__dirname, '..', '..', '.git');
 
-      resolve((stdout && stdout.trim()) || UNKNOWN);
-    });
-  });
+  try {
+    stdout = cp.execSync('git rev-parse HEAD', {cwd: gitDir});
+  } catch (e) {
+    /* ignore, report UNKNOWN */
+  }
 
+  return ((stdout && stdout.toString().trim()) || UNKNOWN);
```

Comment:
Will the `|| UNKNOWN` ever be used, given that `stdout` is initialised to `UNKNOWN` on line 40? Seems like we need one or the other, not both.